crafting-dead-immerse/src/main/java/com/craftingdead/immerse/network/play/SyncBaseScreenMessage.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Crafting Dead
+ * Copyright (C) 2022  NexusNode LTD
+ *
+ * This Non-Commercial Software License Agreement (the "Agreement") is made between you (the "Licensee") and NEXUSNODE (BRAD HUNTER). (the "Licensor").
+ * By installing or otherwise using Crafting Dead (the "Software"), you agree to be bound by the terms and conditions of this Agreement as may be revised from time to time at Licensor's sole discretion.
+ *
+ * If you do not agree to the terms and conditions of this Agreement do not download, copy, reproduce or otherwise use any of the source code available online at any time.
+ *
+ * https://github.com/nexusnode/crafting-dead/blob/1.18.x/LICENSE.txt
+ *
+ * https://craftingdead.net/terms.php
+ */
+
+package com.craftingdead.immerse.network.play;
+
+import com.craftingdead.immerse.client.gui.screen.block.base.ManageBaseMembersView;
+import com.craftingdead.immerse.client.gui.view.ViewScreen;
+import com.craftingdead.immerse.world.level.extension.LegacyBase;
+import com.craftingdead.immerse.world.level.extension.LevelExtension;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import net.minecraft.client.Minecraft;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraftforge.common.util.LogicalSidedProvider;
+import net.minecraftforge.network.NetworkEvent.Context;
+
+public record SyncBaseScreenMessage(UUID base, Map<UUID, String> usernames, ActionType action) {
+
+  public static SyncBaseScreenMessage decode(FriendlyByteBuf in) {
+    return new SyncBaseScreenMessage(in.readUUID(),
+        in.readMap(FriendlyByteBuf::readUUID, FriendlyByteBuf::readUtf),
+        in.readEnum(ActionType.class));
+  }
+
+  public void encode(FriendlyByteBuf out) {
+    out.writeUUID(this.base);
+    out.writeMap(this.usernames, FriendlyByteBuf::writeUUID, FriendlyByteBuf::writeUtf);
+    out.writeEnum(this.action);
+  }
+
+  public boolean handle(Supplier<Context> ctx) {
+    ctx.get().enqueueWork(() -> {
+      switch (action) {
+        case OPEN -> LogicalSidedProvider.CLIENTWORLD
+            .get(ctx.get().getDirection().getReceptionSide())
+            .map(LevelExtension::getOrThrow)
+            .map(LevelExtension::getLandManager)
+            .ifPresent(landManager -> {
+              var land = landManager.getLandOwner(this.base);
+              if (land instanceof LegacyBase legacyBase) {
+                Minecraft.getInstance().setScreen(
+                    ManageBaseMembersView.createScreen(legacyBase, this.usernames));
+              }
+            });
+        case SYNC_MEMBERS, SYNC_AND_CLEAR -> {
+          if (Minecraft.getInstance().screen instanceof ViewScreen view
```

Comment:
This may crash dedicated servers, if it does, move it into a separate static method which will stop `Minecraft` being class loaded

Reply:
I assume when registering a packet, using `NetworkDirection.PLAY_TO_CLIENT` indicates that the packet will only ever be processed by the client, so it should be safe to do any client action there right?

Reply:
Not necessarily as the message class will still be class loaded on the server which could cause a crash.